src/pages/Register.ts

```diff
@@ -76,21 +82,17 @@ const Register = () => {
   const register = async () => {
     try {
       setBusy(true);
-      const orgFields = partOfOrganization
-        ? {
-            institute,
-            department,
-            title,
-          }
-        : {};
-      await Ajax().User.profile.set({
+      const newlyRegisteredUserProfile: TerraUserProfile = {
         firstName: givenName,
         lastName: familyName,
         contactEmail: email,
         interestInTerra,
-        ...orgFields,
-      });
-      authStore.update((state) => ({ ...state, registrationStatus: 'registeredWithoutTos' }));
+        institute,
+        department,
+        title,
+      };
+      await Ajax().User.profile.set(makeSetUserProfileRequest(newlyRegisteredUserProfile));
```

Comment:
Not necessarily for this PR, but it looks like we set the profile here and then immediately fetch it again (in `refreshTerraProfile`). Does the setProfile endpoint not return the profile?

Reply:
You would think it would.. but nope, it just returns a userStatus object https://github.com/broadinstitute/firecloud-orchestration/blob/7041b10131632504c0af4be2c00586e8b96762da/src/main/scala/org/broadinstitute/dsde/firecloud/service/RegisterService.scala#L45C9-L45C9